### PR TITLE
:bug: [#1042] Disallow mutating published ZaakTypen via the admin

### LINKED ISSUE #1042: The admin does not always block mutating published zaaktypen
**General information**
Open Zaak version: 1.4.x / 1.5.x

**Describe the bug**

It's possible via work-arounds to create new Eigenschappen (and possibly also statustypen, rollen...) in the catalogi admin interface for a zaaktype that is already published. This breaks the expectation that zaaktypen can not change after publication, as dictated by the standard.

The API _does_ block this however, this is limited to the admin interface.

**To Reproduce**
1. Create a zaaktype
2. Publish the zaaktype
3. Click the "Toon eigenschappen"
4. Click "Eigenschap toevoegen" in the top right
5. Fill out the details and submit - the eigenschap is now added for a published zaaktype which should not be allowed.

**Expected behavior**

A validation error should prevent the eigenschap from being saved. This should also be checked for all other sub-resources.

### PR DESCRIPTION
**Pull request by stevenbal**
Fixes #1042 

**Changes**

* Disallow mutating published ZaakTypen via the admin
* Fix change form incorrectly displaying as readonly

### CODE REVIEW

File: src/openzaak/components/catalogi/tests/admin/test_eigenschap_admin.py [hunk 1]
```diff
@@ -49,3 +51,42 @@ def test_validation_length_comma_separated(self):
 
         new_specificatie = EigenschapSpecificatie.objects.get()
         self.assertEqual(new_specificatie.lengte, "5,3")
+
+
+@tag("issue-1042")
```
Comment (sergei-maertens) [inline src/openzaak/components/catalogi/tests/admin/test_eigenschap_admin.py:56]:
```suggestion
@tag("gh-1042")
```
let's see if we can stick to the pattern of tags used in other places :P

File: src/openzaak/components/catalogi/tests/admin/test_resultaattype_admin.py [hunk 2]
```diff
@@ -536,3 +540,49 @@ def test_selectielijstklasse_correctly_prefiltered(self, m):
...
+        error = response.html.find("div", {"class": "field-zaaktype"}).find(
+            "ul", {"class": "errorlist"}
+        )
+
+        self.assertEqual(
+            error.text,
+            _("Creating a relation to non-concept {resource} is forbidden").format(
+                resource="zaaktype"
+            ),
+        )
```
Comment (sergei-maertens) [inline src/openzaak/components/catalogi/tests/admin/test_resultaattype_admin.py:587]: isn't it easier to check `response.context["adminform"].errors`?

File: src/openzaak/components/catalogi/tests/admin/test_eigenschap_admin.py [hunk 3]
```diff
@@ -49,3 +51,42 @@ def test_validation_length_comma_separated(self):
...
+class EigenschapAdminTests(WebTest):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = SuperUserFactory.create()
```
Comment (sergei-maertens) [inline src/openzaak/components/catalogi/tests/admin/test_eigenschap_admin.py:60]: _always_ call the super method

File: src/openzaak/components/catalogi/models/mixins.py [hunk 4]
```diff
@@ -119,3 +119,32 @@ class Meta:
...
+            return
+
+        if not getattr(self, self.resource_name).concept:
+            raise ValidationError(
+                {
+                    self.resource_name: _(
+                        "Creating a relation to non-concept {resource_name} is forbidden"
+                    ).format(resource_name=self.resource_name)
+                }
+            )
```
Comment (sergei-maertens) [inline src/openzaak/components/catalogi/models/mixins.py:146]: Can't we implement that as a validator on the FK field instead?

```py
class StatusType(...):
    zaaktype = models.ForeignKey("ZaakType", ..., validators=[validate_concept_status])
```

Comment (stevenbal) [reply src/openzaak/components/catalogi/models/mixins.py:146]: You're right, I wasn't sure about this mixin approach

**Comment by sergei-maertens** [comment]
The failing tests may be because the validators kick in in DRF after all? :thinking:

**Comment by stevenbal** [comment]
> The failing tests may be because the validators kick in in DRF after all? thinking

It seems that the model field validation is run before the serializer validation

File: src/openzaak/components/catalogi/models/eigenschap.py [hunk 5]
```diff
@@ -209,5 +213,8 @@ class Meta:
         verbose_name = _("Eigenschap")
         verbose_name_plural = _("Eigenschappen")
 
+    def clean(self, *args, **kwargs):
```
Comment (sergei-maertens) [inline src/openzaak/components/catalogi/models/eigenschap.py:216]:
```suggestion
    def clean(self):
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
`models.Model.clean` does not accept any args or kwargs

**Pull request merged**